code/__HELPERS/game.dm

```diff
@@ -527,21 +527,16 @@
 
 // Find an obstruction free turf that's within the range of the center. Can also condition on if it is of a certain area type.
 /proc/find_obstruction_free_location(range, atom/center, area/specific_area)
-	var/list/turfs = RANGE_TURFS(range, center)
 	var/list/possible_loc = list()
 
-	for(var/turf/found_turf in turfs)
-		var/area/turf_area = get_area(found_turf)
-
+	for(var/turf/found_turf as anything in RANGE_TURFS(range, center))
```

Comment:
the `as anything` here might be slower honestly, `in block` has some optimizations to it iirc

Reply:
i need to double check

Reply:
RANGE_TURFS() only returns turfs since it uses block() so as anything is faster here

Reply:
RANGE_TURFS is block, and `in block` has a built-in compiler optimization for it, as far as I am aware. It's the same reason we didn't have people do the old typecast vars for this one either.

Reply:
So, to put it simply, I have to remove `as anything`.